tensorboard/webapp/app_routing/deep_link_provider.ts

```diff
@@ -20,11 +20,30 @@ import {State} from '../app_state';
 
 import {SerializableQueryParams} from './types';
 
+/**
+ * The `DeepLinkProvider` interface standardizes the relationship between the
+ * app state and the URL parameters.  It is expected that when either an update
+ * to the URL parameters should change app state, or, in the other direction,
+ * when an update to the state should cause a change in the URL parameters,
+ * that logic should be invoked through a `DeepLinkProvider`.
+ */
 @Injectable()
 export abstract class DeepLinkProvider {
+
+  /**
+   * Modifies the URL to match the state of the app.
```

Comment:
This method is not intended to do the mutation but only grab certain state from the store into query parameters. Also, the resulting `Observable` should emit when the query parameter should change.